### PR TITLE
partial update testing chapter - rc1

### PR DESCRIPTION
**Pull request by mdegani**
- edits up and including 14.5.2

### CODE REVIEW

File: handout/testing/components/async.md [hunk 1]
```diff
@@ -1,8 +1,8 @@
 # Testing Asynchronous Actions
 
-Some components rely on asynchronous actions to work. This can be tricky to test, as we don't know exactly when to look at the DOM for results. Fortunately Angular provides a function `fakeAsync` which essential fakes asynchronous behaviour. FakeAsync will wrap our test in a zone, and then listen for any asynchronous operations, like setTimeouts, Promises, callbacks, etc but will not actually call those functions asynchronously. Instead, it will rely on us calling the `tick` function to call those functions immediately and simulate time elapsing.
```
Comment (mdegani) [inline handout/testing/components/async.md]: The previous page in the book also tests an async action.  We can now use `whenStable()` for the.  FakeAsyc is useful for precise control over the passage of time.

File: handout/testing/components/async.md [hunk 2]
```diff
@@ -26,15 +26,13 @@ class SampleComponent {
...
+it('Should work',
+  fakeAsync(inject([TestComponentBuilder], (tcb: TestComponentBuilder) => {
     tcb.createAsync(SampleComponent).then(fixture => {
-      tick();
-
       fixture.debugElement.nativeElement.querySelector('button')
         .click();
 
-      tick();
+      tick(2000);
```
Comment (mdegani) [inline handout/testing/components/async.md]: The delay is required.

File: handout/testing/components/injecting-dependencies.md [hunk 3]
```diff
@@ -46,39 +46,52 @@ class MockQuoteService {
...
+}
+from '@angular/compiler/testing';
+
 describe('Testing Quote Component', () => {
-  beforeEachProviders(() => {
-    provide(QuoteService: {useClass: MockQuoteService})
-  });
 
-  it('Should get quote', async(inject([TestComponentBuilder], (tcb) => {
-    return tcb.createAsync(QuoteComponent).then(fixture) => {
```
Comment (mdegani) [inline handout/testing/components/injecting-dependencies.md]: `injectAsync` might have required the promise to be returned to know when the async action has completed, but now that we have `async` Angular uses the test zone to make sure all async operations have completed (queues are empty) before wrapping up.  For this reason, we don't have to return.

File: handout/testing/components/injecting-dependencies.md [hunk 4]
```diff
@@ -46,39 +46,52 @@ class MockQuoteService {
...
 
-      fixture.debugElement.componentInstance.getQuote();
+        fixture.componentInstance.getQuote();
 
-      fixture.detectChanges();
-      var compiled = fixture.debugElement.nativeElement;
-      expect(compiled.querySelector('div')).toHaveText('Test Quote');
-    }
-  })));
+        fixture.whenStable().then(() => {
```
Comment (mdegani) [inline handout/testing/components/injecting-dependencies.md]: `whenStable` ensure the promise completes -- this allows us to test async actions when we don't know the amount of time that needs to pass.

**Comment by bennett000** [comment]
Much needed update.  @mdegani do you have a sample directory with the inline examples working?  If so that would be a nice bonus to build into the repo under `examples/testing`. 

Even without the example test this is an essential update.  :shipit:

**Pull request merged**